operator_ui/src/pages/Jobs/generateJobSpecDefinition.ts

```diff
@@ -90,21 +90,83 @@ export const generateJSONDefinition = (
 }
 
 export const generateTOMLDefinition = (
-  jobSpecAttributes: ApiResponse<OcrJobSpec>['data']['attributes'],
+  jobSpecAttributes: ApiResponse<JobSpecV2>['data']['attributes'],
 ): string => {
+  if (jobSpecAttributes.type === 'fluxmonitor') {
+    return generateFluxMonitorDefinition(jobSpecAttributes)
+  }
+
+  if (jobSpecAttributes.type === 'offchainreporting') {
+    return generateOCRDefinition(jobSpecAttributes)
+  }
+
+  return ''
+}
+
+function generateOCRDefinition(
+  attrs: ApiResponse<JobSpecV2>['data']['attributes'],
+) {
   const ocrSpecWithoutDates = {
-    ...jobSpecAttributes.offChainReportingOracleSpec,
+    ...attrs.offChainReportingOracleSpec,
     createdAt: undefined,
     updatedAt: undefined,
   }
 
   return stringifyJobSpec({
     value: {
-      type: 'offchainreporting',
-      schemaVersion: 1,
+      type: attrs.type,
+      schemaVersion: attrs.schemaVersion,
       ...ocrSpecWithoutDates,
-      observationSource: jobSpecAttributes.pipelineSpec.dotDagSource,
-      maxTaskDuration: jobSpecAttributes.maxTaskDuration,
+      observationSource: attrs.pipelineSpec.dotDagSource,
+      maxTaskDuration: attrs.maxTaskDuration,
+    },
+    format: JobSpecFormats.TOML,
+  })
+}
+
+function generateFluxMonitorDefinition(
+  attrs: ApiResponse<JobSpecV2>['data']['attributes'],
+) {
+  const {
+    fluxMonitorSpec,
+    name,
+    pipelineSpec,
+    schemaVersion,
+    type,
+    maxTaskDuration,
+  } = attrs
+
+  // There should always be a flux monitor spec but safeguarding due to type
+  // constraints
+  if (!fluxMonitorSpec) {
+    return ''
+  }
+  const {
+    contractAddress,
+    precision,
+    threshold,
+    absoluteThreshold,
+    idleTimerPeriod,
+    idleTimerDisabled,
+    pollTimerPeriod,
+    pollTimerDisabled,
+  } = fluxMonitorSpec
+
+  return stringifyJobSpec({
+    value: {
+      type,
+      schemaVersion,
+      name,
```

Comment:
Spreading attrs might be more succinct (unless there attrs we don't want)

Reply:
I generally like to be more explicit when the data comes from an external source (in this case it's the API) so that if the API changes, it doesn't, unbeknownst to us, add additional attributes. 

If you want this changed I could do this 

```
  const {
    fluxMonitorSpec,
    pipelineSpec,
    ...jobAttrs
  } = attrs
```